client/k8s.go

```diff
@@ -164,137 +175,79 @@ func (k K8sClient) GetOpenEBSNamespaceMap() (map[string]string, error) {
-	label := "cstorvolume.openebs.io/name" + "=" + name
-	CStorVolumeReplicas, err := k.OpenebsCS.CstorV1().CStorVolumeReplicas("").List(context.TODO(), metav1.ListOptions{LabelSelector: label})
+// GetPVs returns a list of PersistentVolumes based on the values of volNames slice.
+// volNames slice if is nil or empty, it returns all the PVs in the cluster.
+// volNames slice if is not nil or not empty, it return the PVs whose names are present in the slice.
+// labelselector takes the label(key+value) and makes an api call with this filter applied. Can be empty string if label filtering is not needed.
+func (k K8sClient) GetPVs(volNames []string, labelselector string) (*corev1.PersistentVolumeList, error) {
+	pvs, err := k.K8sCS.CoreV1().PersistentVolumes().List(context.TODO(), metav1.ListOptions{LabelSelector: labelselector})
 	if err != nil {
-		return nil, errors.Wrapf(err, "error while getting cStor Volume Replica for volume %s", name)
+		return nil, err
 	}
-	if len(CStorVolumeReplicas.Items) == 0 {
-		// TODO: This came during rebase, this shouldn't be required
-		fmt.Printf("Error while getting cStor Volume Replica for %s, no replicas found\n", name)
+	volMap := make(map[string]corev1.PersistentVolume)
+	for _, vol := range pvs.Items {
+		volMap[vol.Name] = vol
 	}
-	return CStorVolumeReplicas, nil
-}
-
-// NodeForVolume used to get NodeName for the volume from the Kubernetes API
-func (k K8sClient) NodeForVolume(volName string) (string, error) {
-	label := cstortypes.PersistentVolumeLabelKey + "=" + volName
-	podInfo, err := k.K8sCS.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{LabelSelector: label})
-	if err != nil {
-		return "", errors.Wrapf(err, "error while getting target Pod for volume %s", volName)
+	var list []corev1.PersistentVolume
+	if volNames == nil || len(volNames) == 0 {
+		return pvs, nil
 	}
-	if len(podInfo.Items) == 0 {
-		return "", errors.New(fmt.Sprintf("Error invalid number of Pods %d for volume %s", len(podInfo.Items), volName))
+	for _, name := range volNames {
+		if pool, ok := volMap[name]; ok {
+			list = append(list, pool)
+		} else {
+			fmt.Printf("Error from server (NotFound): PV %s not found\n", name)
+		}
 	}
-	return podInfo.Items[0].Spec.NodeName, nil
+	return &corev1.PersistentVolumeList{
+		Items: list,
+	}, nil
 }
 
-// GetcStorPools returns a list CSPIs
-func (k K8sClient) GetcStorPools() (*cstorv1.CStorPoolInstanceList, error) {
-	cStorPools, err := k.OpenebsCS.CstorV1().CStorPoolInstances("").List(context.TODO(), metav1.ListOptions{})
+// GetPVC returns a PersistentVolumeClaim object using the pvc name passed.
+func (k K8sClient) GetPVC(name string, namespace string) (*corev1.PersistentVolumeClaim, error) {
+	pvc, err := k.K8sCS.CoreV1().PersistentVolumeClaims(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 	if err != nil {
-		return nil, errors.Wrapf(err, "Error while getting cspi")
+		return nil, errors.Wrap(err, "error while getting persistent volume")
```

Comment:
This method is returning a pvc
```suggestion
		return nil, errors.Wrap(err, "error while getting persistent volume claim")
```